### ISSUE TITLE
`r/virtual_machine`: add support for deploying an OVF as a template

### ISSUE THREAD

**Issue by dejarikra** [vmware/terraform-provider-vsphere#2114]
### Community Guidelines

- [X] I have read and agree to the [HashiCorp Community Guidelines ](https://www.hashicorp.com/community-guidelines).
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Description

Currently, the provider has the [vsphere_ovf_vm_template](https://registry.terraform.io/providers/hashicorp/vsphere/latest/docs/data-sources/ovf_vm_template) **data** object, with examples on how to use it to create a VM off of it.

However, attempting to use the [vsphere_ovf_vm_template](https://registry.terraform.io/providers/hashicorp/vsphere/latest/docs/data-sources/ovf_vm_template) **data** object  directly **without** a corresponding `vsphere_vm` **resource** doesn't actually result in creating an OVF Template on vSphere. On the other hand, such an independent template could directly be created on vSphere using the HTML client.

For reference, here is a snippet of the **data** object I used.

```
data "vsphere_ovf_vm_template" "eksa_bottlerocket" {
  name  = trimsuffix(reverse(split("/", var.eksa_bottlerocket_ova_url))[0], ".ova")

  resource_pool_id  = vsphere_resource_pool.eksa.id
  host_system_id    = vsphere_host.esxi01.id
  datastore_id      = data.vsphere_datastore.iscsi01_esxi01.id
  folder            = vsphere_folder.eksa_templates.path

  remote_ovf_url            = var.eksa_bottlerocket_ova_url
  allow_unverified_ssl_cert = true

  disk_provisioning = "thin"
  ovf_network_map = {
    "VM Network" : vsphere_distributed_port_group.VM_vcsa01.id
  }
}
```

I understand that **data** objects are NOT used to create actual resources on the target infrastructure. It would be great if we can have a corresponding **resource** object to create an OVF VM Template directly on vSphere.

### Use Case(s)

I'm currently in the process of deploying an EKS Anywhere Cluster on my vSphere environment. As per AWS's [documentation](https://anywhere.eks.amazonaws.com/docs/getting-started/vsphere/vsphere-prereq/#prepare-a-vmware-vsphere-environment), one of the requirements is to have an OVA template already available on vSphere.

### Potential Terraform Provider Configuration

I envision that the `vsphere_ovf_vm_template` **resource** be more or less similar to the [vsphere_ovf_vm_template](https://registry.terraform.io/providers/hashicorp/vsphere/latest/docs/data-sources/ovf_vm_template) **data** object, whose documentation can be found here.

**Comment by tenthirtyam** [comment]
If you need to **create** an OVF template, why not use Packer?

**Comment by tenthirtyam** [comment]
I'm not sure the title - "Add support for deploying an OVF template to vSphere" - is an accurate description of the ask here. Today, the provider does support deploying a virtual machine from either a template as well as from a local or remote OVA/OVF. It sounds more like you are asking for the ability to deploy the OVA/OVF but set it as a template instead of powering it on.

**Comment by dejarikra** [comment]
> I'm not sure the title - "Add support for deploying an OVF template to vSphere" - is an accurate description of the ask here. Today, the provider does support deploying a virtual machine from either a template as well as from a local or remote OVA/OVF. It sounds more like you are asking for the ability to deploy the OVA/OVF but set it as a template instead of powering it on.

Yes, this is exactly what I am trying to do. I will update the title accordingly.

**Comment by tenthirtyam** [comment]
More closely aligned with the existing enhancement item https://github.com/hashicorp/terraform-provider-vsphere/issues/1180.

**Comment by kastl-ars** [comment]
I would really like to see this feature, as it took me quite some time to find out from the big documentation that this is just not possible...

**Comment by alonkaliber** [comment]
> More closely aligned with the existing enhancement item #1180.

That would be extremely helpful!

**Comment by jakauppila** [comment]
This is still needed.
